samples/cpp/benchmark_app/inputs_filling.cpp

```diff
@@ -261,6 +261,27 @@ ov::Tensor create_tensor_random(const benchmark_app::InputInfo& inputInfo,
     return tensor;
 }
 
+ov::Tensor create_tensor_random_4bit(const benchmark_app::InputInfo& inputInfo,
+                                     uint8_t rand_min = std::numeric_limits<uint8_t>::min(),
+                                     uint8_t rand_max = std::numeric_limits<uint8_t>::max()) {
+    auto tensor = ov::Tensor(inputInfo.type, inputInfo.dataShape);
+    auto data = reinterpret_cast<uint8_t*>(tensor.data());
+
+    std::mt19937 gen(0);
+    uniformDistribution<int32_t> distribution(rand_min, rand_max);
+    for (size_t i = 0; i < tensor.get_size(); i++) {
+        uint8_t val = static_cast<uint8_t>(distribution(gen));
+        size_t dst_idx = i / 2;
+        if (i % 2) {
+            data[dst_idx] = (val & 0x0f);
+        } else {
+            data[dst_idx] = data[dst_idx] | ((val & 0x0f) << 4);
```

Comment:
```suggestion
            data[dst_idx] = data[dst_idx] | (val & 0xf0);
```
You resample from the distribution each iteration anyway

Reply:
I just noticed that you request this function to generate the values in [0, 15]. That makes `val & 0xf0` to always be zero. Sorry for the misleading. Can you revert?